src/macros.rs

```diff
@@ -1,3 +1,30 @@
+#[doc(hidden)]
+#[macro_export]
+macro_rules! _sel {
```

Comment:
shouldn't `_` prefix be reserved for unused things? perhaps, something like `sel_impl` would be more appropriate

Reply:
OK, changed the name.